python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -1,11 +1,56 @@
 """Utility functions for sampling."""
 
 from collections import defaultdict
-from typing import Dict, Tuple, Union
+from typing import Dict, List, Tuple, Union
 
 import torch
 
 
+def unique_and_compact_nodes_list(
```

Comment:
This is a general version of unique and compact, just unique_and_compact should be fine.